spring-boot-springdoc/src/main/java/com/baeldung/springdoc/controller/BookController.java

```diff
@@ -0,0 +1,76 @@
+package com.baeldung.springdoc.controller;
+
+import java.util.List;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.baeldung.springdoc.exception.BookNotFoundException;
+import com.baeldung.springdoc.model.Book;
+import com.baeldung.springdoc.repository.BookRepository;
+
+@RestController
+@RequestMapping("/api/book")
+public class BookController {
+
+    @Autowired
+    private BookRepository repository;
+
+    @GetMapping("/{id}")
+    public Book findById(@PathVariable long id) {
+        return repository.findById(id)
+            .orElseThrow(() -> new BookNotFoundException());
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/")
+    @ResponseBody
```

Comment:
Also when using RestController annotation, we don't need ResponseBody annotation.